### ISSUE TITLE
Menus freeze

### ISSUE THREAD

**Issue by fustygerbil** [strukturart/o.map#52]
With v1.9.791 on Nokia 800 Tough, D-pad navigation in the menus causes the interface to freeze (doesn't change or accept input) for cca 15 seconds at a time.

Sorry I don't know how to diagnose what's happening.

**Comment by strukturart** [comment]
When does this happen, when starting the app or at some point while using it?

**Comment by fustygerbil** [comment]
It happens at any time, and in the main menu system only.

The app starts normally. I can scroll, add markers, and interact with markers through the marker option menu. But when I press the center D-pad button from the map display to access the main menus, the page becomes unresponsive---either immediately or after ?one D-pad key is pressed.

After cca 15sec I can usually move around the menus smoothly. Then, if I return to the map and again open the menu, it happens again.

**Comment by strukturart** [comment]
When I try the desktop-version it works fine so far: https://strukturart.github.io/o.map/
And I can't notice any freezing on my N800 either. Do you perhaps have a lot of .gpx files on your device? because these always have to be read in and then listed, which is sometimes a bit slow.

**Comment by fustygerbil** [comment]
I had several GPX tracks and deleting them had no effect, but clearing 450MB cached map data has made the app responsive again. I'm glad, because it's a great app.

Have I understood correctly that tiles are cached in a db? So is this a performance constraint on cache size?

A thought: my phone is running a busybox httpd, so I could add a locally hosted set of tiles to omap_maps.json But I'd love to be able to do the download (to local filesystem) using o.map's nice interface and zoom/bbox logic

**Comment by strukturart** [comment]
In order to be able to use the tiles offline I use: https://github.com/MazeMap/Leaflet.TileLayer.PouchDBCached you can also just use the cached map to test what is available offline. Good to know that a problem occurs at around 400mb, so I can inform the user about it. thanks for testing. 

Do you already know the possibility of setting a marker via QR code?

**Comment by fustygerbil** [comment]
I don't know at what point it occurs as my cache didn't increase incrementally. I did what you [shouldn't do](https://blog.openstreetmap.org/tag/bulk-download/) and bulk downloaded several hundred MB of tiles.

I haven't made use of the QR reading function yet, but I've tested it and it works. Inititally I was failing to read geo URI codes, before success with a GPX code (one for the readme!). Cool :)
